sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageEncryptionHelper.java

```diff
@@ -175,14 +192,25 @@ StorageEncryptionHelper withInfrastructureEncryption() {
      * @return StorageEncryptionHelper
      */
     StorageEncryptionHelper withEncryptionKeyFromKeyVault(String keyVaultUri, String keyName, String keyVersion) {
+        return withEncryptionKeyFromKeyVault(keyVaultUri, keyName, keyVersion, null);
+    }
+
+    public StorageEncryptionHelper withEncryptionKeyFromKeyVault(String keyVaultUri, String keyName, String keyVersion, String identityId) {
         Encryption encryption = getEncryptionConfig(true);
         encryption.withKeySource(KeySource.MICROSOFT_KEYVAULT);
         encryption
+            .withEncryptionIdentity(new EncryptionIdentity().withEncryptionUserAssignedIdentity(identityId))
             .withKeyVaultProperties(
                 new KeyVaultProperties().withKeyVaultUri(keyVaultUri).withKeyName(keyName).withKeyVersion(keyVersion));
         return this;
     }
 
+    public StorageEncryptionHelper withEncryptionKeyFromStorage() {
+        Encryption encryption = getEncryptionConfig(true);
```

Comment:
do we want to rename this method as well?

Reply:
This is in implementation and not exposed to user. Guess it's fine. Personally I prefer staying with `withEncryptionKeyFromStorage` to describe how it's implemented.